includes/class-wc-calypso-bridge-base-admin-menu.php

```diff
@@ -0,0 +1,396 @@
+<?php
+
+use Automattic\Jetpack\Status;
+
+/**
+ * Base Admin Menu file.
+ *
+ * @since   x.x.x
+ * @version x.x.x
+ *
+ * The base admin menu controller for Ecommerce WoA sites.
+ */
+abstract class WC_Calypso_Bridge_Base_Admin_Menu {
+	/**
+	 * Holds class instances.
+	 *
+	 * @var array
+	 */
+	protected static $instances;
+
+	/**
+	 * Whether the current request is a REST API request.
+	 *
+	 * @var bool
+	 */
+	protected $is_api_request = false;
+
+	/**
+	 * Domain of the current site.
+	 *
+	 * @var string
+	 */
+	protected $domain;
+
+	/**
+	 * The CSS classes used to hide the submenu items in navigation.
+	 *
+	 * @var string
+	 */
+	const HIDE_CSS_CLASS = 'hide-if-js';
+
+	/**
+	 * Base_Admin_Menu constructor.
+	 */
+	protected function __construct() {
+		$this->is_api_request = defined( 'REST_REQUEST' ) && REST_REQUEST || isset( $_SERVER['REQUEST_URI'] ) && str_starts_with( filter_var( wp_unslash( $_SERVER['REQUEST_URI'] ) ), '/?rest_route=%2Fwpcom%2Fv2%2Fadmin-menu' );
+		$this->domain         = WC_Calypso_Bridge_Instance()->get_site_slug();
```

Comment:
Is this code being run?

Reply:
Added `parent::__construct();` back so it should be run 🙂